core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts

```diff
@@ -171,7 +176,7 @@ export class SkillEditorStateService {
     return this._skillIsBeingLoaded;
   }
 
-  getAssignedSkillTopicData(): string {
+  getAssignedSkillTopicData(): { [topicName: string]: string } | null {
```

Comment:
Please add a code comment explaining why this can return `null`.

Reply:
If the null is omitted, it would lead to `Type 'null' is not assignable to type '{ [topicName: string]: string; }'` since the variable `assignedSkillTopicData` has been typed as  `{ [topicName: string]: string } | null`.

Reply:
You should add a code comment, that comment should explain it in less technical terms but it should show why logically can this return `null`. When (in what situations) will `assignedSkillTopicData` be `null`?

Reply:
Ok, will do!